bastionzero/bzero
# Add previous container logs and pod events to zip file for send-logs in agent

## Description of the change

Files in the zip archive now include:
 - `kube-agent-pod-events` - events for the agent pod. Requires new RBAC permissions. If agent doesnt have permission we will skip sending these. RBAC permissions added [in backend](https://github.com/bastionzero/webshell-backend/pull/1625) and [in chart](https://github.com/bastionzero/charts/pull/21)
 - `kube-agent-previous.log` - previous container logs. If no previous container exists for the pod then this will be omitted.
 - `kube-agent.log` - current agent container logs

## Testing

To test this i built a custom prod image [send-logs-panic-events](https://us-east-1.console.aws.amazon.com/ecr/repositories/private/238681891460/bctl-agent/_/image/sha256:363aef42a6776ea8011322aa57507b4b3500a6f3ad39b53ef671d65b911c729b/details?region=us-east-1) you can use in your bctl agent deployment. This custom image is built from this branch and additional contains a deliberate panic when processing the first kube rest api request. In order to get pod events new rbac permissions are needed which requires rsyncing bastion changes from https://github.com/bastionzero/webshell-backend/pull/1625 and redeploying the kube agent with: `bzero-dev kube -deployKubeAgent `.

After setup steps to test are:

1. Update your bctl-agent deployment to use `send-logs-panic` image
2. Connect to kube target
3. Send any rest api request to the kube agent i.e `kubectl get pods` <- this will cause the agent container to crash and restart
4. Run send-logs zli command: `zli send-logs --target <target-name>`

The[ logs uploaded](https://s3.console.aws.amazon.com/s3/buckets/bastionzero-dev-customer-logs?region=us-east-1&prefix=July/bastionzero/&showversions=false) should now include both current/previous agent logs as separate files and the previous agent logs should contain the panic. 

**backend branch:** 
**zli branch:** 
**charts branch:** 

#### Ready to run system tests?


## Relevant release note information

Release Notes: Adds previous container logs in agent's send-logs capabilities

## Related JIRA tickets

Relates to JIRA: CWC-2713

## Have you considered the security impacts?

Does this PR have any security impact?


If yes, please explain:
